File: src/chain-observers/classifiers/rc-account-classifier.ts
import { IMaxManabarData } from "./account-classifier";
import { CollectorClassifierBase } from "./collector-classifier-base";

export interface IRcAccount {
  name: string;
  rcManabar: IMaxManabarData;
}

export interface IRcAccountData {
  rcAccounts: Record<string, IRcAccount>;
}

export interface IRcAccountCollectorOptions {
  rcAccount: string;
}

export class RcAccountClassifier extends CollectorClassifierBase<IRcAccountData, void, void, IRcAccountCollectorOptions> {
  public static forOptions(options: IRcAccountCollectorOptions) {
    return {
      class: this, // Intentionally using `this` to refer to the class prototype itself later - even though it is not a class **instance**
      options
    };
  }
}

File: python/workerbee/chain_observers/classifiers/rc_account_classifier.py
from __future__ import annotations

from typing import Any

from ..classifier_results import RcAccountClassifierData
from .collector_classifier_base import CollectorClassifierBase, TRegisterEvaluationContext


class RcAccountClassifier(CollectorClassifierBase[RcAccountClassifierData]):
    """IRcAccountData:
      rc_accounts: dict[str, IRcAccount]

    IRcAccount:
      name: str
      rc_manabar: IMaxManabarData  {current_mana, last_update_time, max}
    """

    @classmethod
    def for_options(cls, options: dict[str, Any]) -> TRegisterEvaluationContext:
        return {"class": cls, "options": options}
